src/pyk/kast/outer_lexer.py

```diff
@@ -60,6 +61,10 @@ class TokenType(Enum):
 class Token(NamedTuple):
     text: str
     type: TokenType
+    start_line: int = 0
+    start_column: int = 0
+    end_line: int = 0
+    end_column: int = 0
```

Comment:
Here I'd just track the start location of the `Token`:
```suggestion
    loc: tuple[int, int]
```

The reason for this:
- The lexer is already pretty complicated.
- This is straightforward to implement (and test), it just requires saving `it.loc` on the first character of the token.
- For most tokens of a given parsed text, neither start, nor end location is interesting, so calculating and storing only one of them saves resources.
- For all tokens other than `BUBBLE`, the end location is `(token.line, token.col + len(token.text) - 1)`, so calculating it in the parser is straightforward.
- `BUBBLE` will be inner-parsed anyways, so the end location there is probably not indispensable.